blog/_posts/2021-07-31-Mitchell-Johnson_Jamie-Schmidt.md

```diff
@@ -0,0 +1,118 @@
+  
+Many Brains, such as Markov Brains, are built from Genomes themselves, and this also occurs in the process function.  This allows for Brains that evolve over time, and the Evaluator can either receive the Brain itself or whatever decision-making process it has built from the Genomes the Brain was given.
+  
+## Evaluators / author: jamie
+-how new evaluator functionality was implemented before we made the switch
+-how/where the functionality lives now, how its better this way
+  
+An Evaluator is, perhaps, the thing most users will spend the most time with for their custom experiments.  In any evolutionary situation, Organisms will need to be evaluated on whatever traits they are supposed to be evolving, and the Evaluator assigns their fitness value from whatever outputs the Brains provide.  
+  
+When coding an Evaluator, the most important thing is to make sure that, for whatever inputs your evaluate function requires, you've got Brains that can supply it.  They're configured in the same way that Brains and Genomes are, by being given a list of input names that correspond to the types that the Evaluator is expecting.
+  
+We did not make many changes to the Evaluator model, aside from making it compatible with DynamicOrgs, which in turn made them significantly easier to customize since every DynamicOrg Evaluator follows the same steps: 
+  1. Get the emp::DataMap
+  2. Get all the input values you want from it
+  3. Calculate fitness from them
+  4. Return fitness
+
+## Tutorial - creating and running Aagos using DynamicOrg / author: mitchell
+-explanation of aagos and how it works
+-writing the evaluator
+-writing the aagos brain
+-putting it all together in the .gen file -> .mabe file
+-running the expiriment
+  
+  1. Write the Evaluator. (almost always necessary)
+      - Set up the config.
+      - Write the fitness calculator in OnUpdate from the emp::DataMap the Organism provides after org.GenerateOutput().
+      - Confirm that the outputs of the Brains in the DynamicOrg are properly configured with the Evaluator. 
+  2. Write the Brain. (often necessary)
+      - Set up the config.
+      - If the Brain needs to be built in a custom fashion, customize the Rebuild() function.
+      - Customize the Process() function in order to generate the desired output. 
+  3. Write the Genome. (very rarely necessary)
+      - Set up the config.
+      - Ensure compatibility with the base class.
+      - Customize the Mutate() function.
+  4. Write the Organism. (shouldn't be necessary except in the direst of circumstances)
+      - Set up the config.
+      - Customize the Initialize() and Mutate() functions.
+    
+
+## Conclusion
+-reiterate how our work greatly simplified the end user expirience using mabe2
```

Comment:
I saw in Mitchell’s abstract some specific quantified details about how much the process is being simplified, that would be great to include!!!